src/emitter.ts

```diff
@@ -1207,17 +1210,57 @@ export function emitWebIdl(webidl: Browser.WebIdl, flavor: Flavor) {
             return `extends ${result} `;
         }
 
+        function hasSequenceArgument(s: Browser.Signature) {
+            function typeIncludesSequence(type: string | Browser.Typed[]): boolean {
+                if (Array.isArray(type)) {
+                    return type.some(t => typeIncludesSequence(t.type))
+                }
+                return type === "sequence" || !!sequenceTypedefMap[type];
+            }
+            return !!s.param && s.param.some(p => typeIncludesSequence(p.type));
+        }
+
+        function filterSignaturesWithSequence(signature: Browser.Signature[]): Browser.Signature[] {
+            return signature.filter(hasSequenceArgument).map(s => {
+                const newParams = s.param!.map(p => {
```

Comment:
similarly, `params` would be an easier name to read, although that would add a couple dozen lines of code changes.